cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -427,6 +427,18 @@ public void initialize(WorkflowContext context) throws Exception {
     }
     spec = GSON.fromJson(context.getWorkflowSpecification().getProperty(Constants.PIPELINE_SPEC_KEY),
                          BatchPipelineSpec.class);
+
+    WorkflowContext workflowContext = getContext();
+    if (workflowContext.getDataTracer(PostAction.PLUGIN_TYPE).isEnabled()) {
```

Comment:
Elsewhere we check if preview is enabled slightly differently https://github.com/cdapio/cdap/blob/df92757fd72d544bbd02d1a8b8f3135d50004f97/cdap-app-templates/cdap-etl/cdap-data-streams-base/src/main/java/io/cdap/cdap/datastreams/SparkStreamingPipelineDriver.java#L100

Instead of duplicating this logic, I think it would be better if we add a `isPreviewEnabled(RuntimeContext context)` method to the PipelineSpec class, and call it in both places.